cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -643,6 +643,10 @@ Declaration Declaration::Sequence(std::vector<Declaration> decls) {
   return out;
 }
 
+bool Declaration::IsValid(ExecFactoryRegistry* registry) const {
+  return registry->GetFactory(this->factory_name).ok() && this->options != nullptr;
```

Comment:
@westonpace should we do a full option verification here? I mean validate for all the supported options? 
I am not quite sure if this is an adequate verification. Since it is not sure if doing that kind of a validation is a kind of redundant thing, I left it for discussion.

Reply:
Understood. I agree, without the plan it is not possible to do the validation properly. Should we create a JIRA to address this concretely in future?

Reply:
You can but I don't know how high priority it would be.  I think I'd want to see more users of Acero directly (and not just via Substrait or through bindings) before we worry too much about adding in utility validation at the declaration layer.